client/src/components/related/RelatedProducts.jsx

```diff
@@ -178,19 +85,19 @@ class RelatedProducts extends React.Component {
     const { selectedProduct, selectProduct, checkCache, updateCache } = this.props;
```

Comment:
I really like the brevity of this, and I also tried it for taking in props to more clearly show what props are expected and assign defaults, but I ran into issues with 'undefined' tripping this up. None of the solutions I found to work with 'undefined' properties worked.

Did you have any issues here? Maybe I can pick your brain, or maybe if you test with more conditions you'll come across some of these issues?

Reply:
Testing trips up unless you pass in those props, if that's what you mean. I've just been mocking up dummy data/functions for tests.